examples/native-external-references-mockshop/functions/types.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  FunctionEventKey,
+  FunctionEventMap,
+  FunctionEventContext,
+} from '@contentful/functions-types';
+
+export type FunctionEventHandler<
+  FunctionEvent extends FunctionEventKey = FunctionEventKey,
+  ContextParameters extends Record<string, any> = Record<string, any>
+> = (
+  event: FunctionEventMap[FunctionEvent]['request'],
+  context: FunctionEventContext<ContextParameters>
+) =>
+  | Promise<FunctionEventMap[FunctionEvent]['response']>
+  | FunctionEventMap[FunctionEvent]['response'];
+
+type InstallationParameters = {
```

Comment:
Nit, as this is user-defined I would move this type just above `Product` to nod hide it between these "complex" types that are not really business logic